winthrop/books/tests/test_book_migrations.py

```diff
@@ -90,3 +94,74 @@ def test_slugs_generated(self):
         assert princeps.slug == \
             slugify('Machiavelli %s %s' % (princeps.short_title, princeps.pub_year))
 
+
+@pytest.mark.last
+class TestMigratePlumToFiggy(TestMigrations):
+
+    app = 'books'
+    migrate_from = '0014_make_book_slugs_unique'
+    migrate_to = '0015_plum_to_figgy'
+    serialized_rollback = True
+
+    # loads a book with the old plum manifest and related canvas with
+    # i.e., most complex example that should be handled correctly
+    fixtures = ['test_plum_figgy']
+
+    @patch('winthrop.books.migrations.0015_plum_to_figgy.iiif')
+    @patch('winthrop.books.migrations.0015_plum_to_figgy.IIIFPresentation')
+    @patch('winthrop.books.migrations.0015_plum_to_figgy.requests')
+    def setUp(self, mockrequests, mockpres, mockiiif):
+        # mock out data fixtures for calls to figgy
+        requestpickle = os.path.join(FIXTURE_DIR, "test_plum_figgy_request.pickle")
```

Comment:
I kind of get why you're using pickled responses here, but I do feel like it makes the tests much more opaque. The redirect response at least should be pretty easy to mock, and would let you test that the migration actually uses the redirect location and not just a plum rather than a figgy url.

Perhaps at least a comment documenting how the pickled fixtures were generated? Or use non-pickled responses so that they are easier to inspect?

Reply:
@rlskoeser I'll document the pickled responses more heavily for the IIIF methods (because those make sense), but now that you mention it, mocking out the request is much easier. I think because I was already in the mindset of needing to "tape record" responses I did that one too.